src/game/lara1gun.c

```diff
@@ -164,27 +164,22 @@ void AnimateShotgun()
             }
         }
     } else {
-        if (ani == AF_SG_AIM && Input.action) {
-            ani++;
-        } else if (ani > AF_SG_AIM && ani < AF_SG_DRAW) {
-            ani++;
-            if (ani == AF_SG_DRAW) {
-                ani = Input.action ? AF_SG_RECOIL : AF_SG_UNAIM;
-            }
-        } else {
-            if (ani == AF_SG_RECOIL) {
+        if (T1MConfig.enable_shotgun_target_fix && Input.action
+            && !Lara.target) {
+            if (ani >= AF_SG_AIM && ani < AF_SG_DRAW) {
+                ani++;
+                if (ani == AF_SG_DRAW) {
+                    ani = AF_SG_RECOIL;
+                }
+            } else if (ani == AF_SG_RECOIL) {
                 if (Input.action) {
```

Comment:
`Input.Action` is checked here for both true and false.

Reply:
I don't see where it's checked for false? Isn't it actually double checking for true again which it already has to be based on `if (T1MConfig.enable_shotgun_target_fix && Input.action && !Lara.target)`? I might be misunderstanding something though since I'm new to the code.

Reply:
I was comparing to the code for `if (!T1MConfig.enable_shotgun_target_fix)`, which doesn't have a big `if` for `Input.Action` and looks like:
```
                if (ani == AF_SG_RECOIL) {
                    if (Input.action) {
                        FireShotgun();
                        ani++;
                    } else {
                        ani = AF_SG_UNAIM;
                    }
                }
```
When we get there in the old code, we're not inside any `if` that checks `Input.Action`, so the `else` case can run too. Guns/arms anim code is not exactly my forte, so I'm a bit confused and not sure right now if we need that :)